modcc/solvers.cpp

```diff
@@ -185,6 +185,11 @@ void SparseSolverVisitor::visit(BlockExpression* e) {
 }
 
 void SparseSolverVisitor::visit(AssignmentExpression *e) {
+    if (conserve_ && !A_[deq_index_].empty()) {
```

Comment:
Double check:
* that `conserve_==true` implies that `A_` has been initialized.
* that putting CONSERVE before or in the middle of the "assignment expressions" won't (a) break things (b) give different linear system.

Reply:
Can `A_` be initialized to `n` rows on construction, when `dvars_` is provided? The only problem might be if `::reset()` is called?

Reply:
* `conserve_==true` does imply that `A_` has been initialized, but I will check the initialization of `A_` before checking `conserve_` anyway
* Putting `CONSERVE` before or in the middle of the assignment expressions won't break things and will give the same linear system; changing the order of the variables in the conserve statement will give a different linear system
* I think it should be okay to initialize `A_` once `dvars_` is filled, I assume the reason it wasn't done like that originally is because of the `::reset()`